### PR TITLE
Bugfix/#544 update subscribers npe

### LINKED ISSUE #544: [2604] odd tick crash
I don't know what the player did to cause this server crash

https://paste.fedoraproject.org/545227/86187349/

### PR DESCRIPTION
**Pull request by cltnschlosser**
Closes #544

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 1]
```diff
@@ -562,16 +562,21 @@ public void onServerTick(@NotNull final TickEvent.ServerTickEvent event)
     /**
      * Update Subscribers with Colony, Citizen, and AbstractBuilding Views.
      */
-    public void updateSubscribers()
+    private void updateSubscribers()
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java:567]: MAJOR The Cyclomatic Complexity of this method "updateSubscribers" is 11 which is greater than 10 authorized. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 2]
```diff
@@ -562,16 +562,21 @@ public void onServerTick(@NotNull final TickEvent.ServerTickEvent event)
...
     {
+        // If the world or server is null, don't try to update the subscribers this tick.
+        if (world == null || world.getMinecraftServer() == null) {
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java]: MINOR Move this left curly brace to the beginning of next line of code. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

File: src/test/java/com/minecolonies/coremod/inventory/InventoryTest.java [hunk 3]
```diff
@@ -42,6 +44,6 @@ public void addStackTest()
         final Item testItem = mock(Item.class);
         final ItemStack stuff = new ItemStack(testItem, 3);
         inventory.setInventorySlotContents(0, stuff);
-        //assertThat(inventory.getStackInSlot(0), is(stuff));
+        assertSame(inventory.getStackInSlot(0), stuff);
```
Comment (MinecoloniesSonar) [inline src/test/java/com/minecolonies/coremod/inventory/InventoryTest.java]: MAJOR Add a message to this assertion. [rule](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2698)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 3 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'
[MINOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/Colony.java (hunk 2 above)
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java]: I think this and then :shipit:

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

**Comment by marvin-bitterlich** [comment]
LGTM! @Raycoms 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/547/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**